hearth/lib/hearth/dns/host_resolver.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Hearth
+  module DNS
+    # Resolve a host name to an address.
+    # @api private
+    class HostResolver
+      def resolve_address(options = {})
+        addrinfo_list = addrinfo(options)
+        ipv6 = ipv6_addr(addrinfo_list, options) if use_ipv6?
```

Comment:
I think that can make sense.

Reply:
Actually, I think the opposite now.. DNS resolution may not want ipv6 or even not want ipv4.

Reply:
I've kept it the way it is. Because HostResolver implementation (getaddrinfo) can dictate family, i.e. INET or INET6, I think it makes sense to have this class also be aware of whether it's supported or not.